core/cap-0034.md

```diff
@@ -23,19 +23,22 @@ chosen `<close time, transaction set>` pair.
 
 With the current protocol, the following sequence of events can occur:
 
-- A transaction `T` with an expiration time makes it into a candidate
-`StellarValue`'s transaction set, with its expiration being no later than the
-`closeTime` in that `StellarValue`.
-- That transaction set is nominated, but with a later `closeTime`, as some
-other candidate(s) had a later `closeTime`, and the current protocol selects
-the maximum.
-- The ledger-closing process begins, and the ledger header is updated to reflect
-the newly-chosen `closeTime`.
-- The ledger-closing process reaches the point of applying transactions.
+1. A transaction `T` with an expiration time makes it into a candidate
+`StellarValue`'s transaction set, with its expiration time being greater than
+(or equal to) the last ledger close time.
+
+2. That transaction set is nominated, but with a `closeTime` greater than
+`T`'s expiration time.  In other words, T has raced into the exact ledger
+in which turned out to have been going to expire.
+
+3. The ledger-closing process begins, and the ledger header is updated to
+reflect the newly-chosen `closeTime`.
+
+4. The ledger-closing process reaches the point of applying transactions.
 Each transaction is charged a fee and consumes its sequence number, and is then
 re-validated (and then applied if and only if the re-validation succeeds).
-`T`'s `closeTime` turns out to be before the newly-chosen `closeTime`.  `T`
-therefore returns `txTOO_LATE` without being applied.
+`T`'s `closeTime` is before the newly-chosen `closeTime`.  `T` therefore returns
+`txTOO_LATE` without being applied.
 
 So `T` consumes a sequence number (and is charged a fee) but is never applied.
```

Comment:
Re-reading this, it still seems confusing to me. The term "executed" is highly ambiguous in this context: I would consider any transaction that was included in a transaction set to have been "executed". The missing detail here is that in the absence of this race condition it is possible to have a chain of transactions each of which is guaranteed to succeed.

Reply:
I am looking for the word which in implementation terms means "has its `TransactionFrame::applyOperations()` method called", because it is the operation(s) of `T` whose postconditions would ensure the preconditions of a transaction that depended on `T`.  What should I say?  (I've tried "applied" in the latest revision, but I don't know whether that's correct either.)

Reply:
(Indeed, in another revision I have started using the full phrase "has its operations applied" in some places.)